cmd/fabric8-jenkins-proxy/main_cors_test.go

```diff
@@ -31,37 +27,54 @@ func TestAPIServerCORSHeaders(t *testing.T) {
 
 	reader, _ := http.NewRequest("POST", "/doesntmatter", nil)
 
-	// Check for origin "https://*.openshift.io"
+	// hope UUID generates valid characters for domain name
 	randomOrigin := uuid.NewV4().String()
-	reader.Header.Set("Origin", "https://"+randomOrigin+".openshift.io")
-	writer := httptest.NewRecorder()
-	apiServer.Handler.ServeHTTP(writer, reader)
-	assert.Equal(t, "https://"+randomOrigin+".openshift.io", writer.Header().Get("access-control-allow-origin"))
 
-	// Check for origin "https://openshift.io"
-	reader.Header.Set("Origin", "https://openshift.io")
-	writer = httptest.NewRecorder()
-	apiServer.Handler.ServeHTTP(writer, reader)
-	assert.Equal(t, "https://openshift.io", writer.Header().Get("access-control-allow-origin"))
+	t.Run("Check for origin https://*.openshift.io", func(t *testing.T) {
+		givenOrigin := "https://" + randomOrigin + ".openshift.io"
+		expectedOrigin := "https://" + randomOrigin + ".openshift.io"
 
-	// Check for origin "https://localhost:*"
-	randomOrigin = uuid.NewV4().String()
-	reader.Header.Set("Origin", "https://localhost:"+randomOrigin)
-	writer = httptest.NewRecorder()
-	apiServer.Handler.ServeHTTP(writer, reader)
-	assert.Equal(t, "https://localhost:"+randomOrigin, writer.Header().Get("access-control-allow-origin"))
+		assertCorsHeaders(reader, givenOrigin, expectedOrigin, apiServer, t)
+	})
 
-	// Check for origin "http://localhost:*"
-	randomOrigin = uuid.NewV4().String()
-	reader.Header.Set("Origin", "http://localhost:"+randomOrigin)
-	writer = httptest.NewRecorder()
-	apiServer.Handler.ServeHTTP(writer, reader)
-	assert.Equal(t, "http://localhost:"+randomOrigin, writer.Header().Get("access-control-allow-origin"))
+	t.Run("Check for origin https://openshift.io", func(t *testing.T) {
+		givenOrigin := "https://openshift.io"
+		expectedOrigin := "https://openshift.io"
+
+		assertCorsHeaders(reader, givenOrigin, expectedOrigin, apiServer, t)
+	})
+
+	t.Run("Check for origin https://localhost:*", func(t *testing.T) {
+		givenOrigin := "https://localhost:" + randomOrigin
+		expectedOrigin := "https://localhost:" + randomOrigin
 
-	// Check that we dont allow a random origin
-	randomOrigin = uuid.NewV4().String()
-	reader.Header.Set("Origin", "https://"+randomOrigin+".io")
-	writer = httptest.NewRecorder()
+		assertCorsHeaders(reader, givenOrigin, expectedOrigin, apiServer, t)
+	})
+
+	t.Run("Check for origin http://localhost:*", func(t *testing.T) {
+		givenOrigin := "http://localhost:" + randomOrigin
+		expectedOrigin := "http://localhost:" + randomOrigin
+
+		assertCorsHeaders(reader, givenOrigin, expectedOrigin, apiServer, t)
+	})
+
+	t.Run("Check that we don't allow a random origin", func(t *testing.T) {
+		givenOrigin := "https://" + randomOrigin + ".io"
+		expectedOrigin := ""
+
+		assertCorsHeaders(reader, givenOrigin, expectedOrigin, apiServer, t)
+	})
+}
+
+func assertCorsHeaders(reader *http.Request, givenOrigin string, expectedOrigin string, apiServer *http.Server, t *testing.T) {
```

Comment:
golang  convention is to use short variable names. e.g. `r *http.Request`, https://talks.golang.org/2014/names.slide#8 

when in doubt, refer golang source